### ISSUE TITLE
Test failure with rspec-mocks 3.12.0 and ruby 3.2dev

### ISSUE THREAD

**Issue by mtasaka** [rspec/rspec-core#2977]
### Subject of the issue
As seen on https://github.com/rspec/rspec-core/actions/runs/3331958744/jobs/5512427869 , rspec-core 3.12.0 test suite is now failing with ruby 3.2dev like:
```
Finished in 9.89 seconds (files took 0.56894 seconds to load)
2209 examples, 10 failures, 4 pending

Failed examples:

rspec ./spec/rspec/core/configuration_options_spec.rb:146 # RSpec::Core::ConfigurationOptions#configure forces force_color
rspec ./spec/rspec/core/configuration_options_spec.rb[1:3:13] # RSpec::Core::ConfigurationOptions#configure sets exclude_pattern before `requires` so users can check `files_to_run` in a `spec_helper` loaded by `--require`
rspec ./spec/rspec/core/configuration_options_spec.rb:152 # RSpec::Core::ConfigurationOptions#configure forces no_color
rspec ./spec/rspec/core/configuration_options_spec.rb:139 # RSpec::Core::ConfigurationOptions#configure forces color
rspec ./spec/rspec/core/configuration_options_spec.rb:102 # RSpec::Core::ConfigurationOptions#configure configures the seed (via `order`) before requires so that required files can use the configured seed
rspec ./spec/rspec/core/configuration_options_spec.rb:95 # RSpec::Core::ConfigurationOptions#configure sets default_path before `files_or_directories_to_run` since it relies on it
rspec ./spec/rspec/core/configuration_options_spec.rb:80 # RSpec::Core::ConfigurationOptions#configure sets default_path before loading specs
rspec ./spec/rspec/core/configuration_options_spec.rb[1:3:12] # RSpec::Core::ConfigurationOptions#configure sets pattern before `requires` so users can check `files_to_run` in a `spec_helper` loaded by `--require`
rspec ./spec/rspec/core/configuration_options_spec.rb:111 # RSpec::Core::ConfigurationOptions#configure configures `only_failures` before `files_or_directories_to_run` since it affects loaded files
rspec ./spec/rspec/core/configuration_options_spec.rb:397 # RSpec::Core::ConfigurationOptions default_path gets set before files_or_directories_to_run
```

### Your environment
* Ruby version:  I am now using https://github.com/ruby/ruby/commit/4b1504ae0a
* rspec-core version: 3.12.0, as well as rspec-mocks, -support, -expectations all 3.12.0

### Steps to reproduce
* install rspec series 3.12.0
* checkout rspec-core git, checkout v3.12.0
* rspec spec/

### Expected behavior
* No failure

### Actual behavior
* Test failure as written above.
* Also, I try building rspec-core with Fedora copr: result is https://copr.fedorainfracloud.org/coprs/mtasaka/rubygem-newruby-test/build/4998834/ , see builder-live.log.gz

### Notes
I think error is raised by this code: https://github.com/rspec/rspec-mocks/commit/e931e818b577172b89fb4583fc336fbcd25df36b : this code is not included in rspec-mocks 3.11.2 (and actually tihs commit is not included in rspec-mocks 3-11-maintenance branch)

**Comment by pirj** [comment]
The failure is quite strange at a glance.

> this code is not included in rspec-mocks 3.11.2 (and actually tihs commit is not included in rspec-mocks 3-11-maintenance branch)

It's from [this PR](https://github.com/rspec/rspec-mocks/pull/1461) and it's [released in 3.12](https://github.com/rspec/rspec-mocks/blob/main/Changelog.md#3120--2022-10-26) and the changes are in the [`3-12-maintenance` branch](https://github.com/rspec/rspec-mocks/blob/f4b10a99f94e33d68c311a8177c7b3b669d6a941/lib/rspec/mocks/error_generator.rb#L272).

`"3.2.0dev" > "3.0.0"` is `true` [here](https://github.com/rspec/rspec-support/blob/49b89c6af0408f543c0be3772553853ec3fc9efd/lib/rspec/support/ruby_features.rb#L132).

Can this be a regression in Ruby?

**Comment by mtasaka** [comment]
Just updated https://github.com/ruby/ruby/commit/6bf458eefd and rspec-core testsuite fails as well: build result here:
https://copr.fedorainfracloud.org/coprs/mtasaka/rubygem-newruby-test/build/5002380/
